---
title: "Expressed Variant Reporting"
author: "Brandon Michael Blobner, Jenny Leopoldina Smith, Ahmad Al Khleifat"
date: "1/8/2021"
output:
  html_document:
    theme: yeti
    highlight: breezedark
    always_allow_html: yes
    toc: true
    toc_float: true
    toc_depth: 3
    number_sections: true
    fig_caption: true

runtime: shiny
---


```{r setup, include=FALSE, echo=FALSE}
require(knitr)
knitr::opts_knit$set(root.dir = here::here())


knitr::opts_chunk$set(tidy.opts=list(width.cutoff=50),
                      tidy=TRUE,
                      fig.align='center',
                      fig.width = 5, fig.height = 7)

getwd()
```

```{r echo=FALSE, message=FALSE}
library(dplyr)
library(magrittr)
library(tidyr)
library(stringr)

library(here)
library(shiny)

library(ggplot2)
library(gridExtra)
library(RColorBrewer)

suppressPackageStartupMessages(library(vcfR))
suppressPackageStartupMessages(library(VariantAnnotation))
```


# About The Test: Variant Detection from RNA-seq


![snpReportR](snpReporter_logo.png)



```{r echo=FALSE, message=FALSE, warning=FALSE}
pt.info <- data.frame("Information"=c("Name:", 
                                        "DOB:",
                                        "Sex:",
                                        "Sample Type:"), 
                        "Value"=c("Jane Doe","01/01/1900","F","RNA"))
dr.info <- data.frame("Information"=c("Name:", 
                                        "Doctor identification number:",
                                        "Hospital:"), 
                        "Value"=c("Dr.X", "12345", "NCI"))
                      
                      
knitr::kable(pt.info, caption = "Patient details:",align="l") %>% 
  kableExtra::kable_paper(full_width = F, position = "left")

knitr::kable(dr.info, caption = "Test ordered by:", align="l") %>% 
  kableExtra::kable_paper(full_width = F, position = "float_left")
```


```{r echo=FALSE, message=FALSE, results="hide"}
#https://gist.github.com/sephraim/3d352ba4893df07a2c35d8f227ab17ac

vcf <-  suppressMessages(vcfR::read.vcfR(here("Data/DRR131561.variants.HC_init.wAnnot.vcf.gz")))

vcf.df <- cbind(as.data.frame(getFIX(vcf)),
                   INFO2df(vcf))

# head(vcf.df)
```

```{r warning=FALSE, echo=FALSE, message=FALSE}
vcf.s4 <- suppressMessages(VariantAnnotation::readVcf(here("Data/DRR131561.variants.HC_init.wAnnot.vcf.gz")))

# vcf.test <- readVcf(here("Data/variants.HC_hard_cutoffs_applied.cancer.vcf"))
# head(vcf.s4)
# str(vcf)
```

```{r echo=F}
header_data <- suppressMessages(VariantAnnotation::header(vcf.s4))
SampleName <- header_data@samples
```

This document will help you to understand the more important findings from a gene variant screening. The common definitions of the type of genetic variants (mutations) are described in the figure and in the table in section `Chromosome and Gene Vizualization of Mutations`. 

While a variant may have been detected, the associations with the variant are not perfectly causal and their complex interactions between biology and the environment.


***

# About the Dataset

```{r echo=FALSE, fig.cap="Table 1. Column names and Descriptions"}
header_info <- info(header_data) %>% 
  as.data.frame()

# class(header_info)
# str(header_info)
# header_info

DT::datatable(header_info, class = 'cell-border stripe', 
              caption="Table 1. Column names and descriptions for the CTAT Mutation Pipeline Results.")
```

```{r echo=FALSE, results='hide'}
functional_annots_names <- header_info["ANN","Description"] %>% 
  gsub("^.+\\'(.+)\\'","\\1",.) %>% 
  str_split(., pattern = "\\|") %>% 
  unlist() %>% 
  gsub("\\s", "", .) 

functional_annots_names <- functional_annots_names[-length(functional_annots_names)]
# functional_annots_names
# length(functional_annots_names) #15
```

In addition there are functional annotations for the variants per  transcript from snpEFF. These include:

* "Annotation_Impact"   
* "Feature_Type"        
* "Transcript_BioType"   
* "Rank"                
* "HGVS.c"
* "HGVS.p"
* "cDNA.pos/cDNA.length"
* "CDS.pos/CDS.length"  
* "AA.pos/AA.length"
* "Distance" 


The top variants were ranked by the following attributes:

* FATHMM predicted pathogenicty or splice adjacent
* genes with larger number of SNVs prioritized
* high or moderate impact on the structure of the gene 
* CADD Score/Polyphen Score (not done yet)


```{r echo=FALSE, results='hide', warning=FALSE}
# https://www.biostars.org/p/226965/
function_annots.df <- data.frame(do.call(rbind, strsplit(as.vector(vcf.df$ANN), split = "\\|")))
function_annots.df <- function_annots.df[,1:45]  #keep only the first 3 transcripts
colnames(function_annots.df) <- paste(functional_annots_names,rep(1:3, each=15), sep="_")


head(function_annots.df)
# dim(function_annots.df)
```

```{r echo=FALSE, results='hide'}
#should be able to download this. 
variants.df <- vcf.df %>% 
  mutate(S4_Vector_IDs=names(rowRanges(vcf.s4))) %>% 
  bind_cols(., function_annots.df) %>% 
  
  mutate(rsID=ifelse(!is.na(RS), paste0("rs", RS), RS)) %>% 
  mutate_at(vars(CHASM_FDR:VEST_FDR), ~as.numeric(.)) %>% 
  group_by(GENE) %>% 
  mutate(Number_SNVs_per_Gene=n()) %>% 
  ungroup() %>% 
  
  dplyr::select(GENE,Number_SNVs_per_Gene, COSMIC_ID,
         rsID,CHROM:ALT,FATHMM,SPLICEADJ,
         TISSUE,TUMOR,
         Annotation_1,Annotation_Impact_1,Feature_Type_1,
         Transcript_BioType_1, 
         coding_DNA_change_1=HGVS.c_1, 
         protein_change_1=HGVS.p_1,
         -ANN, everything(), ANN) %>% 
  dplyr::filter(grepl("PATHOGENIC", FATHMM) | !is.na(SPLICEADJ)) %>%  
  dplyr::filter(grepl("HIGH|MODERATE",Annotation_Impact_1) | !is.na(SPLICEADJ)) %>%


  dplyr::filter(VEST_FDR > 0.5 | is.na(VEST_FDR )) %>%
  dplyr::filter(CHASM_FDR > 0.5 | is.na(CHASM_FDR )) %>%
  arrange(desc(FATHMM), desc(Number_SNVs_per_Gene), Annotation_Impact_1)




head(variants.df)
dim(variants.df) #34
length(unique(variants.df$GENE))
# write.csv(variants.df, "Variants_df.csv", row.names = FALSE)
# table(variants.df$Annotation_Impact_1)
```

```{r echo=FALSE, message=FALSE}
types <- variants.df %>% 
  dplyr::group_by(FATHMM, SPLICEADJ,Annotation_Impact_1) %>% 
  summarize(Number_of_Genes=n()) %>%
  ungroup() 


DT::datatable(types,class = 'cell-border stripe', caption = "Table 2. Summary of Expressed Genetic Variants with potential for Pathegenicity.")
# ggplot(variants.df,aes(x=Classification, Y) )
```

A total of 34 variants across 33 genes were identified that met this criteria for further evaluation. 

***

# HumanMine Annotation for Further Results 

HumanMine annotations for these 34 genes were queried from the HumanMine database API. 

```{r echo=FALSE, results="hide"}
library(InterMineR)
source(here::here("HumanMine_query_functions.R"))
source(here::here("query_get_and_filter_function.R"))
```

```{r echo=FALSE}
genes <- variants.df %>% 
  dplyr::slice(1:5) %>% 
  pull(GENE) %>% 
  unique()

message("Retrieving Annotations for Genes.")  
annots <- query_get_and_filter(gene.names=genes)
```

```{r echo=FALSE}
# str(annots)
# length(annots)
# names(annots)
```

```{r echo=FALSE}
gene.expn.regex <- paste(paste(genes,"Top.Expression", sep="."), collapse = "|")
top.gene.expn <- annots[grep(gene.expn.regex, names(annots))] %>% 
  bind_rows()
  # mutate(GENE=rep(genes, each=5)) %>% 
  # dplyr::select(GENE, everything())
# head(top.gene.expn)

DT::datatable(top.gene.expn, class = 'cell-border stripe', 
              caption="Table 3. Expression of Gene in Various Tissues from Top Variant Hits")
```

```{r  fig.width=10, fig.height=15, echo=FALSE, fig.cap="Expression of Genes with identified variants/SNVs in Normal Tissues."}
gene.expn.val.regex <- paste(paste(genes,"Expression", sep="."), collapse = "|")
top.gene.expn.val <- annots[grep(gene.expn.val.regex, names(annots))] %>% 
  bind_rows()

# head(top.gene.expn.val)
bar.plot <- ggplot(top.gene.expn.val, 
       aes(x=Gene.rnaSeqResults.tissue, 
           y=Gene.rnaSeqResults.expressionScore, 
           fill=Gene.rnaSeqResults.tissue)) +
  geom_col() +
  coord_flip() +
  facet_wrap(~Gene.symbol) +
  theme_classic() +
  labs(x="Tissue", y="Expression Score", title="Expression in Normal Tissues") +
  theme(panel.border = element_rect(color="black", fill=NA), 
        strip.text = element_text(size=16, face="bold"), 
        axis.text.y = element_text(angle = 25, hjust=1, vjust=1,size=10),
        axis.text.x = element_text(angle = 25, hjust=1, vjust=1), 
        legend.position = "top",
        legend.text = element_text(size=16))
  
bar.plot
```



References:
*Smith RN, et al. InterMine: a flexible data warehouse system for the integration and analysis of heterogeneous biological data. Bioinformatics. 2012 Dec 1;28(23):3163-5.*

***

# Results Table {.tabset}

Included here are two tables for different gene types. Coding refers to genes that produce proteins, while non-coding refers to genes which do get utilized to produce proteins.   


## Coding Genes

```{r echo=FALSE}
DT::datatable(dplyr::filter(variants.df,
                     Transcript_BioType_1 == "Coding") %>%
                dplyr::select(GENE:protein_change_1,protein_change_1) %>%
                dplyr::slice(1:10),
               class = 'cell-border stripe', 
              caption = "Table 4. Protein Coding Variants, possibly pathogenic.")
```

## Non-Coding Genes

```{r echo=FALSE}
DT::datatable(dplyr::filter(variants.df,
                     Transcript_BioType_1 == "Noncoding") %>%
                dplyr::select(GENE:protein_change_1,protein_change_1),
              class = 'cell-border stripe', 
              caption = "Table 5. Non-Coding Variants, possibly pathogenic")
```

***

# Chromosome and Gene Vizualization of Mutations

```{r echo=FALSE, message=FALSE}
suppressPackageStartupMessages(library(Gviz))
suppressPackageStartupMessages(library(GenomicRanges))
options(ucscChromosomeNames=FALSE)
source(here::here("Gviz_tracks.R"))
```

```{r echo=FALSE}
knitr::kable(data.frame("Type"=c("coding:",
                                 "5prime_UTR:",
                                 "3prime_UTR:",
                                 "introninc:", 
                                  "splice:",
                                 "synonymous_variant:",
                                 "missense_variant:", 
                                 "start/stop:"), 
                        "Description"=c("Mutation is within a coding region",
                                        "Mutation is within 5’ untranslated region",
                                        "Mutation is with 3’ untranslated region",
                                        "Mutation is with an intron region", 
                                        "Mutation is within proximity to a splice-site.",
                                        "Synonymous variant is a mutation in an exon that results in same amino acid (changed codon)", 
                                        "Missense variant is a mutation in an exon that results in a different amino acid (changed codon)",
                                        "Mutation is within a start/stop codon.")), 
             caption = "Table 6. Types of Expressed Variants", align = "l") %>% 
  kableExtra::kable_paper("hover", full_width = F, position = "float_left")
```



***



```{r message=FALSE,fig.width=5, fig.height=7, echo=FALSE, fig.show=TRUE, fig.cap="Figure 1. Percentage of different types of mutations identified."}
donut.df <- variants.df %>% 
  group_by(Annotation_1) %>% 
  summarize(N=n()) %>% 
  mutate_at(vars(Annotation_1), ~case_when(
    grepl("5_prime_UTR", .) ~ "5_prime_UTR_variant", 
    grepl("missense_variant", .) ~ "missense_variant", 
    grepl("splice", .) ~ "splice_site", 
    TRUE ~ .
  )) %>% 
  group_by(Annotation_1) %>% 
  summarise(N=sum(N)) %>% 
  ungroup() %>% 
  mutate(Percent=round(N/sum(N) *100, digits = 2)) %>% 
  arrange(N) %>% 
  mutate(lab.pos = cumsum(Percent)-.5*Percent) %>% 
  mutate(Annotation_1 = factor(Annotation_1, levels=rev(unique(Annotation_1))))


temp <- filter(donut.df, Percent > 5)
donut.types <- ggplot(data = donut.df, 
       aes(x = 2, y = Percent, fill = Annotation_1))+
  geom_col(width = 1.0, color="black", size=0.25) +
  # scale_x_discrete(limits = c(" ", 2)) +
  xlim(0.3, 2.5) +
  annotate(geom="label",
           x=2.0, y=temp$lab.pos,
           label=paste0(temp$Percent,"%"),
           size=3) +
  ggrepel::geom_text_repel(data=filter(donut.df, Percent <= 5),
                           aes(x = 2.5, y = lab.pos,label=paste0(Percent,"%")),
                    nudge_x = 0.25,
                    segment.size = .5,
                    min.segment.length = 0.1,
                    size=3,
                    show.legend = FALSE,
                   inherit.aes = F) +
  labs(title="Variant Locations within Genes") +
  coord_polar("y", start=1) +
  scale_fill_brewer(palette="Paired") +
  theme_void() +
  theme(legend.position = "left", 
        legend.title = element_blank(), 
        legend.text = element_text(size=8))

donut.types
```

*** 

```{r echo=FALSE, message=FALSE, results="hide"}
# txdb <- GenomicFeatures::makeTxDbFromGFF(here::here("GenomicRefs/gencode.v22.chr_patch_hapl_scaff.annotation.gtf.gz"))
# AnnotationDbi::saveDb(txdb, "gencode.v22.chr_patch_hapl_scaff.annotation.sqlite")
txdb <- AnnotationDbi::loadDb(here::here("GenomicRefs/gencode.v22.chr_patch_hapl_scaff.annotation.sqlite"))
GRCh38.txs <- GenomicFeatures::transcripts(txdb)
# txdb
```

```{r echo=FALSE, message=FALSE,results="hide"}
top5.coding <- dplyr::filter(variants.df, 
                     Transcript_BioType_1 == "Coding") %>%  
  dplyr::slice(1:5) %>% #Not accurate. Needs to be top5 unique genes. will deal with later. 
    pull(GENE)

top5.coding <- dplyr::filter(variants.df, Transcript_BioType_1 == "Coding") %>% 
    dplyr::filter(GENE %in% top5.coding) %>% 
    pull(S4_Vector_IDs)
```

```{r fig.width=9, fig.show=TRUE, echo=FALSE, fig.subcap="A vizualization of the SNVs", message=FALSE, warning=FALSE}
gene_tracks(vcf_s4 = vcf.s4[top5.coding], transriptsGR = GRCh38.txs)
```


***


# Expression of Genes with Mutations 

boxplots/violin plots 

DE genes by condition

Sommelier results: Haplyotype/PCA

```{r}

```




***

# What Does This Result Mean for you? And What's Next?

Genetic tests sometimes reveal information that could be relevant to your family such as a health risk that might run in the family, or that family relationships are different from what you expected. 

Can you please add this message in the next steps section, if the report show an association with a gene. Please contact your doctor and a genetic counselor. A genetic counselor can help you understand:
<br>

  1. how your family members may be affected if the test shows a serious health condition runs in your family.
  2. the risk of you and your partner passing on a health condition to your children
your options if you have a child with an inherited health condition and you do not want your next child to inherit it
  3. genetic counsellor can also direct you to relevant patient support group


```{css echo=FALSE}
.watch-out {
  background-color: white;
  border: 3px solid red;
  font-weight: bold;
}
```

```{r class.source="watch-out"}
# Need diagram here
```


***

# More information 

```{r echo=FALSE}
gene.pub.regex <- paste(paste(genes,"Recent.Publications", sep="."), collapse = "|")
recent.pubs <- annots[grep(gene.pub.regex, names(annots))] %>% 
  bind_rows() %>% 
  mutate(GENE=rep(genes, each=5)) %>% 
  dplyr::select(GENE, everything())

# head(recent.pubs)
# dim(recent.pubs)
# lapply(recent.pubs, dim)

DT::datatable(recent.pubs, class = 'cell-border stripe', 
              caption="Table 5. Recent Publications for Top Variant Hits")
```

## Potential Drug Targets 

```{r}
#add drug target data information
```

Can search for additional drugs that may target the mutant genes online using [Drug Targetor](http://drugtargetor.com/) and at [Drug Gene Interaction Database](https://www.dgidb.org/)


## Potential Clinical Trials 

```{r}
#search API at https://clinicaltrials.gov/api/ 
```

## Web resources

```{r}
# Add QR code
# where does it go?
```

***

# Consent for Analysis

```{r echo=FALSE}
message("Please Sign Form. Acknowledment of informed consent is provided upon signing.")
```

```{r echo=FALSE}
patient_signature <- function(input){

  shinyApp(

    ui = fluidPage(

        # Copy the line below to make a text input box
        textInput("text", label = h3("Acknowledgement"), value = "Enter First and Last Name Here"),

        hr(),
        fluidRow(column(3, verbatimTextOutput("value")))
    ),

    server = function(input, output) {

      # You can access the value of the widget with input$text, e.g.
      output$value <- renderPrint({ input$text })
    },
    # options = list(height = 500)
  )
}
```

```{r}
patient_signature()
```

***


# Quality Control 

Embed the IGV output from CTAT Mutation pipeline. 

```{r}
#embed file here
```



***

# Genome References and Software

```{css echo=FALSE}
.boxBorder {
     border: 2px solid black;
     padding: 10px;
     outline: black solid 5px;
     outline-offset: 5px;
}
```

<div class="boxBorder">

Additional information about the pipelines used, check out the github repositories listed below:

  * [CTAT Mutation Pipeline for Input VCF](https://github.com/collaborativebioinformatics/expressed-variant-impact)
  * [Association between Genes and Drug Targets](https://github.com/collaborativebioinformatics/viravate2)
  * [Mixed Sample Graphs QC](https://github.com/collaborativebioinformatics/mixed-sample-graphs)
  * [snpReportR Generation](https://github.com/collaborativebioinformatics/expressed-variant-reporting)
<br>
</div>


<br><br><br><br>

```{r echo=FALSE}
#Should not hardcode the different annotation algorithms used. 
#but for now its just kind a difficult format to deal with. 
header.dat.names <- names(header_data@header@listData)
keep.dat <- grep("SnpEff|format|Command",header.dat.names)
keep.dat <- header_data@header@listData[keep.dat]
keep.dat <- as.data.frame(lapply(keep.dat,function(x) as.data.frame(x)))

DT::datatable(keep.dat, class = 'cell-border stripe', caption = "Table 7. Software and Version Numbers.")
```

```{r echo=FALSE}
message("Information about the Transcript Annotation Database.")
txdb
```



# Session Information

```{r}
sessionInfo()
```

